src/state.cairo

```diff
@@ -70,77 +71,89 @@ pub struct Transaction {
     pub lock_time: u32,
 }
 
-/// Represents an input of a transaction.
+/// Input of a transaction.
 /// https://learnmeabitcoin.com/technical/transaction/input/
-///
-/// NOTE that `txid` and `vout` fields can be resolved via Utreexo set using the TXO index.
 #[derive(Drop, Copy)]
 pub struct TxIn {
     /// The signature script which satisfies the conditions placed in the txo pubkey script
     /// or coinbase script that contains block height (since 227,836) and miner nonce (optional).
     pub script: @ByteArray,
     /// The sequence number of the input.
     pub sequence: u32,
-    /// The reference to the previous output that is being used as an input.
+    /// The reference to the previous output that is being spent by this input.
     pub previous_output: OutPoint,
     /// The witness data for transactions.
-    pub witness: @ByteArray
+    /// A list of items (of different size) pushed onto stack before sig script execution.
+    pub witness: Span<ByteArray>,
```

Comment:
What exactly is witness here? Is it the input that goes into the VM prior to iterating through the script opcodes?

Reply:
Yes, basically for every item in the `witness` array there is an implicit `OP_PUSH`